content/PreparedStatement/PreparedStatement.q.md

```diff
@@ -0,0 +1,1707 @@
+
+- It sets to add new records into the `students` table.
+
+- It replaces the values in the SQL statement with question marks `?`.
+
+- It uses question marks `?` as parameters to be inserted into the `students` table.
+
+- It modifies the values to be inserted into the `students` table with positional placeholders `?`.
+
+- It uses positional placeholders `?` to associate the values to be inserted into the `students` table.
+
+
+/// type=MS, answer=[2,3,4,5]
+
+In `student.php`, which statements correctly describe `$pstmt = $conn->prepare("INSERT INTO students (first_name, last_name, birth_date, gender) VALUES (?,?,?,?)");` on line 10?
+
+- It executes the SQL statement.
+
+- It prepares the SQL statement for execution.
+
+- It returns the result set as a `PDOStatement` object.
+
+- It calls the `prepare()` method of the `$conn` object.
+
+- It assigns the `PDOStatement` object to the variable `$pstmt`.
+
+
+/// type=MS, answer=[3,5]
+
+Which statements correctly describe the `foreach` construct on line 11 of `student.php`?
+
+- It displays all elements in `$row`.
+
+- It accesses through each element in `$data`.
+
+- It iterates through each element in the array.
+
+- It creates and initializes the array variable `$data`.
+
+- It assigns the value of the current element on each iteration to `$row`.
+
+
+/// type=SS, answer=[2]
+
+In `student.php`, what does the `execute()` method do on line 12?
+
+- It executes the SQL statement.
+
+- It executes the argument `$row`.
+
+- It fetches the remaining row of the result set.
+
+- It duplicates the values of `$data` to the SQL statement.
+
+- It assigns the values of the multidimensional array `$data` to the variable `$row`.
+
+
+/// type=SS, answer=[4]
+
+What value is assigned to the variable `$row` on line 12 of `student.php`?
+
+- The SQL statement.
+
+- The DSN parameters.
+
+- The `PDOStatment` object.
+
+- The elements of the multidimensional array `$data`.
+
+- The values of each key of `student1` and `student2`.
+
+
+/// type=MS, answer=[2,3,4]
+
+Which statements correctly describe `$pstmt->execute($row);` on line 12 of `student.php`?
+
+- It queries the SQL statement.
+
+- It returns the boolean value `true`.
+
+- It calls the `execute()` method of the `$pstmt` object.
+
+- It executes the argument `$row` in the `execute()` method.
+
+- It assigns the returned value of the `execute()` method to the variable `$row`.
+
+
+/// type=SS, answer=[3]
+
+In `student.php`, what value is returned by `!$pstmt->execute($row)` on line 12?
+
+- The SQL statement.
+
+- The boolean value `true`.
+
+- The boolean value `false`.
+
+- The `PDOStatement` object.
+
+- The elements of the multidimensional array `$data`.
+
+
+/// type=SS, answer=[3]
+
+Which statement best describes the code on lines 12, 13, and 14 of `student.php`?
+
+- The statement displays the values assigned to the `$row` variable.
```

Comment:
This is correct but we have to be consistent with the rest of the content so we need to rewrite it. Thanks.

the `$row` variable ---> __the variable `$row`__